gui/src/app/components/TabWidget.tsx

```diff
@@ -15,8 +15,17 @@ const TabWidget: FunctionComponent<TabWidgetProps> = ({ labels, children }) => {
   }
 
   const [index, setIndex] = useState(0);
+  const [tabsThatHaveBeenViewed, setTabsThatHaveBeenViewed] = useState<
+    number[]
+  >([]);
```

Comment:
the initial value needs to be `[index]` to get rendering correct on the first go-round